tmail-backend/jmap/extensions-api/src/main/scala/com/linagora/tmail/james/jmap/contact/EmailAddressContact.scala

```diff
@@ -92,22 +109,32 @@ object MinAutoCompleteInputLength {
 case class MinAutoCompleteInputLength(value: Int)
 
 trait EmailAddressContactSearchEngine {
-  def index(accountId: AccountId, fields: ContactFields): Publisher[EmailAddressContact]
+  def index(accountId: AccountId, fields: ContactFields): Publisher[EmailAddressContact] =
+    index(accountId, fields, None)
+
+  def index(accountId: AccountId, fields: ContactFields, addressBookCardUid: Option[String]): Publisher[EmailAddressContact]
```

Comment:
In [this issue comment](https://github.com/linagora/tmail-backend/issues/1683#issuecomment-2772148007) , I understand that Benoit suggested using `addressBookId` as the name.  
However, I’m concerned it might be confusing, since the real case we handle is: same email address appearing in different `contact_id` (or `cardUid` in the ICS file).  
So I chose to name it `addressBookCardUid`.  
Feel free to suggest a better name if you have one.

![image](https://github.com/user-attachments/assets/2115c140-45e2-47e5-bab8-7b84491ae62d)


![image](https://github.com/user-attachments/assets/28ea9d3c-afef-48bd-b9b8-5a682f3f7263)